.github/workflows/collector-builder.yml

```
@@ -57,7 +56,16 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        arch: [amd64, ppc64le, s390x, arm64]
+        include:
+        - arch: amd64
+          runner: ubuntu-24.04
+        - arch: ppc64le
+          runner: ubuntu-24.04
+        - arch: s390x
+          runner: ubuntu-24.04
+        - arch: arm64
+          runner: ubuntu-24.04-arm
+    runs-on: ${{ matrix.runner }}
```

Alternatively, you could do something like this:
```suggestion
        arch: [amd64, ppc64le, s390x, arm64]
    runs-on: ${{ matrix.arch != 'arm64' && 'ubuntu-24.04' || 'ubuntu-24.04-arm' }}
```

---

I tried it your way and it seemed like the expression evaluation didn't work.

---

Reopening this one since I think it would simplify a lot of the logic.

Checked in Falco's repo, since I know they do it this way, looks like I missed a couple of parenthesis, should be:

```yaml
    runs-on: ${{ (matrix.arch != 'arm64' && ubuntu-24.04) || 'ubuntu-24.04-arm' }}
```

https://github.com/falcosecurity/libs/blob/24539f5cdd818aec2498db5c1e961a8dacc4c36c/.github/workflows/ci.yml#L19

---

Cool -- yeah that collapses a lot of logic, will try that again.

---

Ok, got this approach to work which simplifies things.